airflow/utils/process_utils.py

```diff
@@ -79,7 +86,7 @@ def signal_procs(sig):
             else:
                 raise
 
-    if pgid == os.getpgid(0):
+    if IS_WINDOWS and pgid == os.getpid() or not IS_WINDOWS and pgid == os.getpgid(0):
```

Comment:
I think this needs some parentheses? At least for readability.

Reply:
Parentheses would be redundant as the "and" operation precedes "or" according to Boolean Algebra. (would be like a*b+c*d -> (a*b)+(c*d))